explainability-api/microservice/checklist.py

```diff
@@ -0,0 +1,179 @@
+# add functionality to process json to be injected into the db
+import json
+import logging
+import requests
+import itertools
+
+from typing import List
+
+
+def create_query(skill, test_cases: List):
+    """
+    Creates a query and make it suitable for sending to for prediction
+
+    Args:
+        skill: input skill for which the checklist tests are run
+        test_cases (list) : Test cases as a list
+
+    Returns:
+        json_object (json object) : A json object containing the test case and its prediction
+        answer (str) : Prediction for test case made by the skill
+
+    """
+    skill_type = skill["skill_type"]
+    base_model = skill["default_skill_args"].get("base_model")
+    adapter = skill["default_skill_args"].get("adapter")
+    # extract all tests
+    all_tests = [tests["test_cases"] for tests in test_cases]
+    # all_tests = list(itertools.chain.from_iterable([tests["test_cases"] for tests in test_cases]))
+    questions, contexts, answers = list(), list(), list()
+
+    test_type = list(itertools.chain.from_iterable([[test["test_type"]] * len(test["test_cases"])
+                                                    for test in test_cases]))
+    capability = list(itertools.chain.from_iterable([[test["capability"]] * len(test["test_cases"])
+                                                    for test in test_cases]))
+    test_name = list(itertools.chain.from_iterable([[test["test_name"]] * len(test["test_cases"])
+                                                    for test in test_cases]))
+
+    for tests in all_tests:
+        questions.append([query["question"] for query in tests])
+        # list of list for mcq else list
+        contexts.append([query["context"] if skill_type != "multiple-choice"
+                         else query["context"] + "\n" + "\n".join(query["options"])
+                         for query in tests])
+        answers.extend([query.get("answer") if "answer" in query.keys() else query.get("prediction_before_change")
+                        for query in tests])
+
+        # TODO
+        # send batch to the skill query endpoint
+
+    prediction_requests = list()
+    # create the prediction request
+    for idx in range(len(questions)):
+        for question, context in zip(questions[idx], contexts[idx]):
+            request = dict()
+            request["num_results"] = 1
+            request["user_id"] = "ukp"
+            request["skill_args"] = {"base_model": base_model, "adapter": adapter, "context": context}
+            request["query"] = question
+            prediction_requests.append(request)
+
+    model_inputs = dict()
+    model_inputs["request"] = prediction_requests
+    model_inputs["answers"] = answers
+    model_inputs["test_type"] = test_type
+    model_inputs["capability"] = capability
+    model_inputs["test_name"] = test_name
+    # logger.info("inputs:", model_inputs)
```

Comment:
remove commented code

Reply:
Use comments only when you feel that some part of the code is hard to understand and commenting will improve readability.